actor/cfnetworkingaction/policy.go

```diff
@@ -0,0 +1,184 @@
+	var allWarnings Warnings
+
+	applications, warnings, err := actor.V3Actor.GetApplicationsBySpace(spaceGUID)
+	allWarnings = append(allWarnings, Warnings(warnings)...)
+	if err != nil {
+		return []Policy{}, allWarnings, err
+	}
+
+	var v1Policies []cfnetv1.Policy
+	v1Policies, err = actor.NetworkingClient.ListPolicies()
+	if err != nil {
+		return []Policy{}, allWarnings, err
+	}
+
+	var policies []Policy
+	emptyPolicy := Policy{}
+	for _, v1Policy := range v1Policies {
+		policy := transformPolicy(applications, v1Policy)
+		if policy != emptyPolicy {
+			policies = append(policies, transformPolicy(applications, v1Policy))
+		}
+	}
+
+	return policies, allWarnings, nil
+}
+
+func (actor Actor) NetworkPoliciesBySpaceAndAppName(spaceGUID string, srcAppName string) ([]Policy, Warnings, error) {
+	var allWarnings Warnings
+	var appGUID string
+
+	applications, warnings, err := actor.V3Actor.GetApplicationsBySpace(spaceGUID)
+	allWarnings = append(allWarnings, Warnings(warnings)...)
+	if err != nil {
+		return []Policy{}, allWarnings, err
+	}
+
+	var v1Policies []cfnetv1.Policy
+
+	srcApp, warnings, err := actor.V3Actor.GetApplicationByNameAndSpace(srcAppName, spaceGUID)
+	allWarnings = append(allWarnings, Warnings(warnings)...)
+	if err != nil {
+		return []Policy{}, allWarnings, err
+	}
+
+	appGUID = srcApp.GUID
+	v1Policies, err = actor.NetworkingClient.ListPolicies(appGUID)
+	if err != nil {
+		return []Policy{}, allWarnings, err
+	}
+
+	var policies []Policy
+	emptyPolicy := Policy{}
+	for _, v1Policy := range v1Policies {
+		if appGUID == "" || v1Policy.Source.ID == appGUID {
+			policy := transformPolicy(applications, v1Policy)
+			if policy != emptyPolicy {
+				policies = append(policies, transformPolicy(applications, v1Policy))
+			}
+		}
+	}
+
+	return policies, allWarnings, nil
+}
+
+func (actor Actor) RemoveNetworkPolicy(spaceGUID, srcAppName, destAppName, protocol string, startPort, endPort int) (Warnings, error) {
+	var allWarnings Warnings
+
+	srcApp, warnings, err := actor.V3Actor.GetApplicationByNameAndSpace(srcAppName, spaceGUID)
+	allWarnings = append(allWarnings, Warnings(warnings)...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	destApp, warnings, err := actor.V3Actor.GetApplicationByNameAndSpace(destAppName, spaceGUID)
+	allWarnings = append(allWarnings, Warnings(warnings)...)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	policyToRemove := cfnetv1.Policy{
+		Source: cfnetv1.PolicySource{
+			ID: srcApp.GUID,
+		},
+		Destination: cfnetv1.PolicyDestination{
+			ID:       destApp.GUID,
+			Protocol: cfnetv1.PolicyProtocol(protocol),
+			Ports: cfnetv1.Ports{
+				Start: startPort,
+				End:   endPort,
+			},
+		},
+	}
+
+	v1Policies, err := actor.NetworkingClient.ListPolicies(srcApp.GUID)
+	if err != nil {
+		return allWarnings, err
+	}
+
+	for _, v1Policy := range v1Policies {
+		if v1Policy == policyToRemove {
+			return allWarnings, actor.NetworkingClient.RemovePolicies([]cfnetv1.Policy{policyToRemove})
```

Comment:
instead of getting list of policies and trying to find it in order to remove it, better approach would be to just delegate that to `RemovePolicies` method on networking client. It would make 1 http call instead of 2. And it is not going to be vulnerable to race conditions where `ListPolicies` has it and `RemovePolicies` will actually fail to remove it. `RemovePolicies`. Or the other way around. `RemovePolicies` can just throw a corresponding error when API returns 404.

Reply:
Our API will return a successful response when the policy being deleted does not exist. Is it desired/required that the CLI return an error in this situation? If so then we should probably leave this as is, but I would lean towards not returning an error at all.

Reply:
@jaydunk usually CLI delete commands do not fail when object they are trying to delete does not exist, they will print a warning to user only.

Reply:
Yeah it looks like that is the current behavior, we create an error in the actor which is caught in the command and converted to a warning. This was the recommendation of the CLI team.

Reply:
It might not be possible to throw the error/display the warning if the API does not error when deleting non-existant objects. We should let Dies know about this during acceptance.